### ISSUE TITLE
MSB3030 Build Error

### ISSUE THREAD

**Issue by josuegomes** [Microsoft/vcpkg#910]
Windows 7 Professional, Visual Studio 2015 Professional Update 3

Steps to reproduce (Windows 7 Pro, Visual Studio 2015 Pro Upd 3)

1. cd c:\
2. mkdir projects
3. cd projects
4. git ... vcpkg
5. cd vcpkg
6. bootstrap-vcpkg.bat
7. vcpkg install boost
   (cmake, and boost installed)
8. vcpkg integrate install
9. Launch Visual Studio 2015 Upd 3
10. File, New, Project
11. Installed, Templates, Visual C++, Win32, Win32 Console Application
12. Finish
13. Select All and clear
14. Paste this

```
#include "stdafx.h"

#include <iostream>
#include <boost/date_time/posix_time/posix_time.hpp>

int main()
{
	boost::posix_time::ptime now = boost::posix_time::second_clock::local_time();
	std::cout << boost::posix_time::to_simple_string(now) << "\n";
	return 0;
}
```

15. Press F5, or Ctrl-F5, or Ctrl-Shift-B
16. In the Output widnow the error:

`C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(3963,5): error MSB3030: Could not copy the file "ost_date_time-vc140-mt-gd-1_63.dll" because it was not found.
`

There were no errors when building Boost. See attached logs.

[boost-x86-windows.zip](https://github.com/Microsoft/vcpkg/files/905776/boost-x86-windows.zip)

**Comment by josuegomes** [comment]
After increasing MSBuild verbosity I found this is triggered by the path length. I was creating the project under my Documents folder thus:

```
1>                     C:\users\jgo\documents\visual studio 2015\Projects\ConsoleApplication1\Debug\bo (TaskId:46)
1>                     ost_date_time-vc140-mt-gd-1_63.dll (TaskId:46)
```

MSBuild is only getting the last line "ost_date_time-vc140-mt-gd-1_63.dll"


Tried another path with no success:

```
1>                     C:\Projects\VS2015\ConsoleApplication1\Debug\boost_date_time-vc140-mt-gd-1_63.d (TaskId:33)
1>                     ll (TaskId:33)

```

Only when I shrink the project name and total path length is really small I got no errors:

```
1>                     C:\Projects\ConApp1\Debug\boost_date_time-vc140-mt-gd-1_63.dll (TaskId:32)
1>                     Output Item(s): ReferenceCopyLocalPaths=C:\Projects\ConApp1\Debug\boost_date_time-vc140-mt-gd-1_63.dll (TaskId:32)

```

**Comment by ras0219-msft** [comment]
Hi @josuegomes, thanks for reporting this issue and providing such a detailed diagnosis 👍.

After following the steps you listed on my machine (Win10, VS2015U3@14.0.25431.01+VS2017), everything appears to work as expected; the output doesn't appear truncated at all:
```
1>Target "AppLocalFromInstalled" in file "D:\src\vcpkg\scripts\buildsystems\msbuild\vcpkg.targets" from project "c:\Users\roschuma\documents\visual studio 2015\Projects\ConsoleApplication1\ConsoleApplication1\ConsoleApplication1.vcxproj" (target "RegisterOutput" depends on it):
1>       Task "WriteLinesToFile"
1>       Done executing task "WriteLinesToFile".
1>       Using "Exec" task from assembly "Microsoft.Build.Tasks.Core, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
1>       Task "Exec"
1>         powershell.exe -ExecutionPolicy Unrestricted -noprofile -File "D:\src\vcpkg\scripts\buildsystems\msbuild\applocal.ps1" "c:\users\roschuma\documents\visual studio 2015\Projects\ConsoleApplication1\Debug\ConsoleApplication1.exe" "D:\src\vcpkg\installed\x86-windows\debug\bin" "Debug\ConsoleA.8B0B96FD.tlog\ConsoleApplication1.write.1u.tlog"
1>         C:\users\roschuma\documents\visual studio 2015\Projects\ConsoleApplication1\Debug\boost_date_time-vc141-mt-gd-1_63.dll
1>       Done executing task "Exec".
1>       Task "Exec" skipped, due to false condition; ('$(VcpkgConfiguration)' == 'Release') was evaluated as ('Debug' == 'Release').
1>     1>
1>Done building target "AppLocalFromInstalled" in project "ConsoleApplication1.vcxproj".
```

I'll try to reproduce this on a win7 machine on Monday.

**Comment by josuegomes** [comment]
@ras0219-msft if you need help to reproduce, test fixes or collect logs please let me know.

**Comment by alexkaratarakis** [comment]
Fixed with 92c0a91d3fdd9a9df5d3feae611981ea170703f5.
Please pull and try rebuilding your project (no need to rebuild `vcpkg`).

**Comment by josuegomes** [comment]
Fixed.
